app/controllers/auth/PayeAuthAction.scala

```diff
@@ -43,10 +44,11 @@ class PayeAuthActionImpl @Inject()(override val authConnector: AuthConnector,
         HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
 
       authorised(ConfidenceLevel.L200 and AuthNino(hasNino = true) and CredentialStrength(CredentialStrength.strong))
-        .retrieve(Retrievals.nino) {
-          case Some(nino) => {
+        .retrieve(Retrievals.externalId and Retrievals.nino) {
+          case Some(userId) ~ Some(nino) => {
             block {
               PayeAuthenticatedRequest(
+                userId,
```

Comment:
Don't have the ticket but can we check why we need this external ID?

Reply:
I don't think we need to audit externalId . Looked into the splunk and couldn't find out any advantage of having this field as we are already logging NINO that is unique to the user and we are going to have sessionIDs as well in audit logs.

Reply:
This was requested by the analyst building the dashboard, as it makes it equivalent to the payload for the SA part of the service.